src/adsp_config.c

```diff
@@ -237,19 +263,50 @@ static void parse_str_key(const toml_table_t *table, struct parse_ctx *ctx, cons
 
 	len = strlen(temp_s);
 	if (len > capacity) {
-		*error = log_err(-EINVAL, "Too long input for key '%s' (%d > %d)\n", key, len,
-				 capacity);
+		if (len > 20) {
+			static const char ellipsis[] = "...";
+			const size_t el_len = sizeof(ellipsis);
+
+			strncpy(temp_s + len - el_len, ellipsis, el_len);
+		}
+
+		*error = log_err(-EINVAL, "Too long input '%s' for key '%s' (%d > %d) characters\n",
+				 temp_s, key, len, capacity);
```

Comment:
I guess this part should truncate `temp_s` to 20 (17) chars and add `...` suffix.  Then it should be `strncpy(temp_s + max_len - el_len,....`

Reply:
> I guess this part should truncate temp_s to 20 (17) chars and add ... suffix.

I don't understand, that's what the code above does.

What is `max_len`?

Reply:
Fixed, thanks @ktrzcinx for the thorough review.